code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/filter/ServiceCreateValidationFilter.java

```diff
@@ -49,6 +60,30 @@ public Object create(String type, ApiRequest request, ResourceManager next) {
                     "name");
         }
 
+        // 2. if a part of sidekick, scale should match other sidekick participants scale
+        Map<String, Object> launchConfig = DataAccessor
```

Comment:
Do we really need to add validate this.  From a user perspective this will be annoying.  If the value must be X, can we just make it X?  For this scale.  We should try to keep these value in sync on scale up/scale down.  But the backend should assume they are.  So if they get out of sync because of some race condition, things shouldn't fail.  On the backed just take the max value.  (Maybe you do this already, haven't gotten that far yet in the code)